src/lib/containers/synapse_form_wrapper/SynapseForm.tsx

```diff
@@ -375,8 +378,8 @@ export default class SynapseForm extends React.Component<
         this.state.currentStep,
         this.state.steps,
       )
-      if (this.formRef.current) {
-        this.formRef.current.submit()
+      if (this.submitButtonRef.current) {
+        this.submitButtonRef.current.click()
       }
```

Comment:
For some unknown reason, `formRef.current.submit` stopped working upon upgrading React (in both v17 and v18). Invoking `click` on the submit button fixes it.